k2/csrc/ragged_ops_inl.h

```diff
@@ -94,9 +94,13 @@ Ragged<T> Stack(int32_t axis, int32_t num_srcs, Ragged<T> **src) {
   // context compatibility and num-axes compatibility.
   RaggedShape ans_shape = Stack(0, num_srcs, src_shapes.data());
   Array1<T> ans_values = Append(num_srcs, src_values.data());
-  Ragged<T> ans(ans_shape, ans_values);
-  if (axis == 1) ans = Transpose(ans);
-  return ans;
+  if (axis == 1) {
```

Comment:
@csukuangfj we should not change the implementation of `Stack` to call `MakeTransposable`, when `axis == 1`, the input shapes of `Stack` are suppose to have same `Dim0()`. 

https://github.com/k2-fsa/k2/blob/9cf4b14245cecb07e5d43297f9553caeb8ca8a24/k2/csrc/ragged_ops.h#L126-L128

So there are some bugs in `TopSort` that go agaist this requirements, we need to fix that bug instead of the code in `Stack` . BTW we definitely need to check the code in `TopSort`  to **make sure if we really need to call `MakeTransposable` or not**.) Anyway, you can leave the bug for now and I can fix them later.